src/mplhep/plot.py

```diff
@@ -541,17 +541,25 @@ def hist2dplot(
 
     plt.sca(ax)
 
-    text_artists = []
-    if labels is not None:
+    _labels: np.ndarray | None = None
+    if (type(labels) == bool) or (np.iterable(labels)):
```

Comment:
```suggestion
    if isinstance(labels, bool) or (np.iterable(labels)):
```

I remember @henryiii telling me to always use `isinstance` instead of `type` to make sure things are checked correctly.